player.go

```diff
@@ -0,0 +1,493 @@
+package spotify
+
+import (
+	"bytes"
+	"encoding/json"
+	"net/http"
+	"net/url"
+	"strconv"
+)
+
+// PlayerDevice contains information about a device that a user can play music on
+type PlayerDevice struct {
+	// ID of the device. This may be empty.
+	ID string `json:"id"`
+	// IsActive If this device is the currently active device.
+	IsActive bool `json:"is_active"`
+	// IsRestricted Whether controlling this device is restricted. At present if
+	// this is "true" then no Web API commands will be accepted by this device.
+	IsRestricted bool `json:"is_restricted"`
+	// Name The name of the device.
+	Name string `json:"name"`
+	// Type Device type, such as "Computer", "Smartphone" or "Speaker".
+	Type string `json:"type"`
+	// Volume The current volume in percent.
+	Volume int `json:"volume_percent"`
+}
+
+// PlayerDeviceList is a list of devices
+type PlayerDeviceList struct {
+	PlayerDevices []PlayerDevice `json:"devices"`
+}
+
+// PlayerState contains information about the current playback.
+type PlayerState struct {
+	CurrentlyPlaying
+	// Device The device that is currently active
+	Device PlayerDevice `json:"device"`
+	// ShuffleState Shuffle is on or off
+	ShuffleState bool `json:"shuffle_state"`
+	// RepeatState off, track, context
+	RepeatState string `json:"repeat_state"`
+}
+
+// Context is the playback context
+type Context struct {
+	// ExternalURLs of the context, or null if not available.
+	ExternalURLs map[string]string `json:"external_urls"`
+	// Endpoint of the context, or null if not available.
+	Endpoint string `json:"href"`
+	// Type of the item's context. Can be one of album, artist or playlist.
+	Type string `json:"type"`
+	// URI of the context.
+	URI URI `json:"uri"`
+}
+
+// CurrentlyPlaying contains the information about currently playing items
+type CurrentlyPlaying struct {
+	// Timestamp when data was fetched
+	Timestamp int `json:"timestamp"`
+	// Context current context
+	Context Context `json:"context"`
+	// Progress into the currently playing track.
+	Progress int `json:"progress_ms"`
+	// IsPlaying If something is currently playing.
+	IsPlaying bool `json:"is_playing"`
+	// The currently playing track. Can be null.
+	Item *FullTrack `json:"Item"`
+}
+
+type Offset struct {
+	//Position is zero based and can’t be negative.
```

Comment:
Surprised the linter doesn't pick that up.

RE Units: The api doesn't specify, in the context of their example it's the offset of a track in a playlist or list of uris, but I suspect it's also milliseconds when given a track.